### DISCUSSION TITLE
Still getting rate-limited.

### DISCUSSION THREAD

**Discussion by ghost** [Ximaz/instagram-downloader#2] [Q&A] [answered]
Hello,
I wondered why do I still getting ratelimited by the Instagram API ? I tried to switch between my router and my 4G box, but I still stuck. Any idea ?

**Comment by Ximaz** [comment, marked as answer]
Hi @SergentCoder
First of all, thanks for opening this discussion, it might help some people about what's happening. In a recent commit, I specified I increased the delay between requests of 5 seconds, which makes a new delay of 25 seconds + a random number in range of ``[1;3[`` .

It seems it's not enough for Instgram to trust. I search for some more informations and found some cookies from headers when I'm logged in Instagram and doing request. This project aims at avoiding the login phase, but it seems that those cookies are delivered only to logged in people.

I didn't searched any further for now, but I will dig this ASAP to see if those cookie can't be generated in any way.

The only thing I can recommand for now is to use VPN when you're blocked. I don't think I will implement a proxy way in this project since Instagram is constantly banning new proxies.

I hope I answered your question. If you have any left, feel free to ask.

**Reply by ghost** [reply]
Yeah, you answered my question, thanks. I hope there is a way to hijacked those cookies.
